// @stellar/open-rpc-docs-react/dist/ArrayFieldItemTemplate/ArrayFieldItemTemplate.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
/** The `ArrayFieldItemTemplate` component is the template used to render an items of an array.
 *
 * @param props - The `ArrayFieldTemplateItemType` props for the component
 */
function ArrayFieldItemTemplate(props) {
    const { children, className, disabled, hasToolbar, hasMoveDown, hasMoveUp, hasRemove, hasCopy, index, onCopyIndexClick, onDropIndexClick, onReorderClick, readonly, registry, uiSchema, } = props;
    const { CopyButton, MoveDownButton, MoveUpButton, RemoveButton } = registry.templates.ButtonTemplates;
    const btnStyle = {
        flex: 1,
        paddingLeft: 6,
        paddingRight: 6,
        fontWeight: 'bold',
    };
    return (react_1.default.createElement("div", { className: className + ' row' },
        react_1.default.createElement("div", { className: hasToolbar ? 'col col--9' : 'col col--12' }, children),
        hasToolbar && (react_1.default.createElement("div", { className: 'col col--3 array-item-toolbox', style: {
                display: 'flex',
                justifyContent: 'center',
                alignItems: 'center',
            } },
            (hasMoveDown || hasMoveUp) && (react_1.default.createElement(MoveUpButton, { style: btnStyle, disabled: disabled || readonly || !hasMoveUp, onClick: onReorderClick(index, index - 1), uiSchema: uiSchema, registry: registry })),
            (hasMoveUp || hasMoveDown) && (react_1.default.createElement(MoveDownButton, { style: btnStyle, disabled: disabled || readonly || !hasMoveDown, onClick: onReorderClick(index, index + 1), uiSchema: uiSchema, registry: registry })),
            hasCopy && (react_1.default.createElement(CopyButton, { style: btnStyle, disabled: disabled || readonly, onClick: onCopyIndexClick(index), uiSchema: uiSchema, registry: registry })),
            hasRemove && (react_1.default.createElement(RemoveButton, { style: btnStyle, disabled: disabled || readonly, onClick: onDropIndexClick(index), uiSchema: uiSchema, registry: registry }))))));
}
exports.default = ArrayFieldItemTemplate;
