// @stellar/open-rpc-docs-react/dist/InteractiveMethod/InteractiveMethod.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const validator_ajv8_1 = __importDefault(require("@rjsf/validator-ajv8"));
const traverse_1 = __importDefault(require("@json-schema-tools/traverse"));
const core_1 = __importDefault(require("@rjsf/core"));
const ArrayFieldTemplate_1 = __importDefault(require("../ArrayFieldTemplate/ArrayFieldTemplate"));
const ArrayFieldItemTemplate_1 = __importDefault(require("../ArrayFieldItemTemplate/ArrayFieldItemTemplate"));
const FieldErrorTemplate_1 = __importDefault(require("../FieldErrorTemplate/FieldErrorTemplate"));
const FieldTemplate_1 = __importDefault(require("../FieldTemplate/FieldTemplate"));
const ObjectFieldTemplate_1 = __importDefault(require("../ObjectFieldTemplate/ObjectFieldTemplate"));
const qs = require('qs');
const { useHistory, useLocation } = require('@docusaurus/router');
const uiSchema = {
    'ui:description': '',
    "ui:submitButtonOptions": {
        "norender": true,
    }
};
function AddButton(props) {
    const { icon, iconType, ...btnProps } = props;
    return (react_1.default.createElement("button", { ...btnProps, className: btnProps.className + " button button--primary", type: 'button' }, "+"));
}
function RemoveButton(props) {
    const { icon, iconType, ...btnProps } = props;
    const style = {
        ...btnProps.style,
        minWidth: '35px',
        maxWidth: '36px',
        border: undefined
    };
    return (react_1.default.createElement("button", { ...btnProps, style: style, className: "button button--outline button--primary", type: 'button' }, "-"));
}
function MoveUpButton(props) {
    const { icon, iconType, ...btnProps } = props;
    const style = {
        ...btnProps.style,
        minWidth: '35px',
        maxWidth: '36px',
    };
    return (react_1.default.createElement("button", { ...btnProps, style: style, className: "button button--outline button--primary", type: 'button' }, "\u25B2"));
}
function MoveDownButton(props) {
    const { icon, iconType, ...btnProps } = props;
    const style = {
        ...btnProps.style,
        minWidth: '35px',
        maxWidth: '36px',
    };
    return (react_1.default.createElement("button", { ...btnProps, style: style, className: "button button--outline button--primary", type: 'button' }, "\u25BC"));
}
const InteractiveMethodParam = (props) => {
    const { param, refref } = props;
    const [metamaskInstalled, setMetamaskInstalled] = react_1.default.useState(false);
    const schema = (0, traverse_1.default)(param.schema, (s) => {
        if (typeof s === 'boolean') {
            return s;
        }
        s.description = undefined;
        s.summary = undefined;
        return s;
    }, { mutable: false });
    schema.title = undefined;
    (0, react_1.useEffect)(() => {
        const installed = !!window?.ethereum;
        setMetamaskInstalled(installed);
    }, []);
    return (react_1.default.createElement(core_1.default, { schema: schema, formData: props.formData, showErrorList: false, uiSchema: uiSchema, validator: validator_ajv8_1.default, ref: refref, disabled: !metamaskInstalled, templates: {
            ArrayFieldItemTemplate: ArrayFieldItemTemplate_1.default,
            ArrayFieldTemplate: ArrayFieldTemplate_1.default,
            FieldErrorTemplate: FieldErrorTemplate_1.default,
            FieldTemplate: FieldTemplate_1.default,
            ButtonTemplates: { AddButton, RemoveButton, MoveUpButton, MoveDownButton },
            ObjectFieldTemplate: ObjectFieldTemplate_1.default
        }, onChange: props.onChange, liveValidate: metamaskInstalled }));
};
const InteractiveMethod = (props) => {
    const history = useHistory();
    const queryString = qs.parse(history.location.search, {
        ignoreQueryPrefix: true,
        decoder: (value, defaultEncoder, _, type) => {
            if (type === "key") {
                return defaultEncoder(value);
            }
            if (/^(\d+|\d*\.\d+)$/.test(value)) {
                return parseFloat(value);
            }
            const keywords = {
                true: true,
                false: false,
                null: null,
                undefined,
            };
            if (value in keywords) {
                return keywords[value];
            }
            try {
                return decodeURIComponent(value);
            }
            catch (e) {
                return value;
            }
        },
    });
    const { method, components, selectedExamplePairing } = props;
    const [requestParams, setRequestParams] = react_1.default.useState(queryString || {});
    const [executionResult, setExecutionResult] = react_1.default.useState();
    const [metamaskInstalled, setMetamaskInstalled] = react_1.default.useState(false);
    const formRefs = method.params.map(() => (0, react_1.createRef)());
    (0, react_1.useEffect)(() => {
        if (!selectedExamplePairing || Object.keys(queryString).length > 0) {
            return;
        }
        const defaultFormData = selectedExamplePairing?.params.reduce((memo, exampleObject, i) => {
            const ex = exampleObject;
            memo[method.params[i].name] = ex.value;
            return memo;
        }, {});
        setRequestParams(defaultFormData);
    }, [selectedExamplePairing]);
    const handleChange = (change, i) => {
        setRequestParams((val) => {
            const newVal = {
                ...val,
                [method.params[i].name]: change.formData,
            };
            history.replace({
                search: qs.stringify(newVal, { encode: false }),
            });
            return newVal;
        });
    };
    const methodCall = {
        method: method.name,
        params: method.paramStructure === "by-name" ? requestParams : method.params.map((p, i) => requestParams[p.name] || undefined),
    };
    const handleExec = async () => {
        // loop over refs
        formRefs.forEach((formRef) => {
            formRef.current.validateForm();
        });
        try {
            const response = await window.ethereum.request(methodCall);
            setExecutionResult(response);
        }
        catch (e) {
            setExecutionResult(e);
        }
    };
    const jsCode = `await window.ethereum.request(${JSON.stringify(methodCall, null, "  ")});`;
    (0, react_1.useEffect)(() => {
        const installed = !!window?.ethereum;
        setMetamaskInstalled(installed);
    }, []);
    return (react_1.default.createElement("div", { className: "container" },
        !metamaskInstalled &&
            react_1.default.createElement("div", { className: "row" },
                react_1.default.createElement("div", { className: "alert alert--danger", role: "alert" },
                    "Install MetaMask for your platform and refresh the page. The interactive features in this documentation require installing ",
                    react_1.default.createElement("a", { target: "_blank", href: "https://metamask.io/download/" }, "MetaMask"),
                    "."),
                react_1.default.createElement("br", null)),
        method.params.length > 0 &&
            react_1.default.createElement(react_1.default.Fragment, null,
                react_1.default.createElement("div", { className: "row" },
                    react_1.default.createElement("h3", null, "Params"),
                    react_1.default.createElement("div", { className: "col col--12" }, method.params.map((p, i) => (react_1.default.createElement(react_1.default.Fragment, null,
                        react_1.default.createElement("h4", null, p.name),
                        react_1.default.createElement(InteractiveMethodParam, { refref: formRefs[i], formData: requestParams[p.name], onChange: (change) => handleChange(change, i), param: p })))))),
                react_1.default.createElement("br", null)),
        react_1.default.createElement("div", { className: "row" },
            react_1.default.createElement("h3", null, "Request"),
            react_1.default.createElement("div", { className: "col col--12" },
                components && components.CodeBlock && react_1.default.createElement(components.CodeBlock, { className: "language-js" }, jsCode),
                !components?.CodeBlock &&
                    react_1.default.createElement("pre", null,
                        react_1.default.createElement("code", null, jsCode)))),
        executionResult !== undefined && react_1.default.createElement("div", { className: "row" },
            react_1.default.createElement("h3", null, "Response"),
            react_1.default.createElement("div", { className: "col col--12" },
                components && components.CodeBlock && react_1.default.createElement(components.CodeBlock, { className: "language-json" }, JSON.stringify(executionResult, null, '  ')),
                !components?.CodeBlock &&
                    react_1.default.createElement("pre", null,
                        react_1.default.createElement("code", null, JSON.stringify(executionResult, null, '  '))))),
        react_1.default.createElement("div", { className: "row" },
            react_1.default.createElement("button", { className: "button button--primary button--block", onClick: handleExec, disabled: !metamaskInstalled }, "Send Request"))));
};
exports.default = InteractiveMethod;
