// @stellar/open-rpc-docs-react/dist/ExamplePairing/ExamplePairing.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const lodash_1 = __importDefault(require("lodash"));
const MarkdownDescription_1 = __importDefault(require("../MarkdownDescription/MarkdownDescription"));
class ExamplePairing extends react_1.Component {
    render() {
        const { examplePairing, paramStructure, methodName, uiSchema, components } = this.props;
        if (lodash_1.default.isUndefined(examplePairing)) {
            return null;
        }
        if (lodash_1.default.isUndefined(methodName)) {
            return null;
        }
        const params = paramStructure === 'by-name'
            ? examplePairing.params.reduce((memo, p) => {
                memo[p.name] = p.value;
                return memo;
            }, {})
            : examplePairing.params.map((p) => p.value);
        const methodCall = {
            jsonrpc: '2.0',
            id: 8675309,
            method: methodName,
            params,
        };
        if (Object.keys(params).length === 0) {
            delete methodCall.params;
        }
        const resultObj = {
            jsonrpc: '2.0',
            id: 8675309,
            result: examplePairing.result.value,
        };
        const serverUrl = methodName.indexOf('_') > -1
            ? 'https://platform-server.exampleanchor.com'
            : 'https://soroban-testnet.stellar.org';
        const curlCode = `curl -X POST \\
-H 'Content-Type: application/json' \\
-d '${JSON.stringify(methodCall, null, 2)}' \\
${serverUrl} | jq`;
        const jsCode = `let requestBody = ${JSON.stringify(methodCall, null, 2)}
let res = await fetch('${serverUrl}', {
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
  },
  body: JSON.stringify(requestBody),
})
let json = await res.json()
console.log(json)`;
        const pythonCode = `import json, requests
res = requests.post(${serverUrl}, json=${JSON.stringify(methodCall, null, 4)})
print(json.dumps(res.json(), indent=4))`;
        return (react_1.default.createElement("div", null,
            react_1.default.createElement("div", null,
                react_1.default.createElement(MarkdownDescription_1.default, { uiSchema: uiSchema, source: examplePairing.description || '', className: "example-pairing-description" })),
            react_1.default.createElement("div", null,
                react_1.default.createElement("div", null,
                    react_1.default.createElement("h3", null, "Request"),
                    react_1.default.createElement("div", null,
                        components &&
                            components.CodeBlock &&
                            components.Tabs &&
                            components.TabItem && (react_1.default.createElement(components.Tabs, { groupId: "rpc-examples" },
                            react_1.default.createElement(components.TabItem, { value: "curl", label: "cURL" },
                                react_1.default.createElement(components.CodeBlock, { className: "language-bash" }, curlCode)),
                            react_1.default.createElement(components.TabItem, { value: "javascript", label: "JavaScript" },
                                react_1.default.createElement(components.CodeBlock, { className: "language-js" }, jsCode)),
                            react_1.default.createElement(components.TabItem, { value: "python", label: "Python" },
                                react_1.default.createElement(components.CodeBlock, { className: "language-python" }, pythonCode)),
                            react_1.default.createElement(components.TabItem, { value: "json", label: "JSON" },
                                react_1.default.createElement(components.CodeBlock, { className: "language-json" }, JSON.stringify(methodCall, null, 2))))),
                        !components?.CodeBlock && (react_1.default.createElement("pre", null,
                            react_1.default.createElement("code", null, curlCode)))))),
            react_1.default.createElement("div", null,
                react_1.default.createElement("div", null,
                    react_1.default.createElement("h3", null, "Result"),
                    react_1.default.createElement("div", null,
                        components && components.CodeBlock && (react_1.default.createElement(components.CodeBlock, { className: "language-json" }, JSON.stringify(resultObj, null, '  '))),
                        !components?.CodeBlock && (react_1.default.createElement("pre", null,
                            react_1.default.createElement("code", null, JSON.stringify(resultObj)))))))));
    }
}
exports.default = ExamplePairing;
