// @stellar/open-rpc-docs-react/dist/Method/Method.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const MarkdownDescription_1 = __importDefault(require("../MarkdownDescription/MarkdownDescription"));
const react_1 = __importDefault(require("react"));
const Links_1 = __importDefault(require("../Links/Links"));
const Tags_1 = __importDefault(require("../Tags/Tags"));
const Params_1 = __importDefault(require("../Params/Params"));
const ContentDescriptor_1 = __importDefault(require("../ContentDescriptor/ContentDescriptor"));
const ExamplePairings_1 = __importDefault(require("../ExamplePairings/ExamplePairings"));
const Errors_1 = __importDefault(require("../Errors/Errors"));
const lodash_1 = __importDefault(require("lodash"));
const defaultJsonViewOptions = {
    theme: "summerfruit:inverted",
    collapseStringsAfterLength: 25,
    displayDataTypes: false,
    displayObjectSize: false,
    indentWidth: 2,
    name: false,
};
const Method = ({ method, uiSchema, key, methodPlugins, reactJsonOptions, onExamplePairingChange, components }) => {
    reactJsonOptions = { ...defaultJsonViewOptions, ...reactJsonOptions };
    if (!method) {
        return null;
    }
    if (lodash_1.default.isEmpty(method)) {
        return null;
    }
    const links = method.links;
    return (react_1.default.createElement("div", { id: method.name, key: key, className: "method margin-bottom--sm" },
        react_1.default.createElement("h1", { key: method.name, className: "method-name", style: { marginRight: "3px" } }, method.name),
        method.tags && method.tags.length > 0 && react_1.default.createElement("section", { key: "tags" },
            react_1.default.createElement(Tags_1.default, { tags: method.tags })),
        method.description ?
            react_1.default.createElement("section", { key: "description" },
                react_1.default.createElement(MarkdownDescription_1.default, { uiSchema: uiSchema, source: method.description, className: "method-description" }))
            : method.summary ?
                react_1.default.createElement("p", { key: method.summary, className: "method-summary" }, method.summary)
                : null,
        react_1.default.createElement("section", { key: "params" },
            react_1.default.createElement("div", { style: { marginBottom: "var(--ifm-heading-margin-bottom)" } },
                react_1.default.createElement("h2", { style: { display: 'inline', marginRight: '3px' } }, "Params"),
                react_1.default.createElement("span", null,
                    "(",
                    method.params?.length || 0,
                    ")")),
            method.params && method.params.length > 0 &&
                react_1.default.createElement(Params_1.default, { params: method.params, uiSchema: uiSchema })),
        method.result &&
            react_1.default.createElement("section", { key: "result-title" },
                react_1.default.createElement("div", { style: { marginBottom: "var(--ifm-heading-margin-bottom)" } },
                    react_1.default.createElement("h2", { style: { display: 'inline', marginRight: '3px' } }, "Result"),
                    react_1.default.createElement("span", null,
                        react_1.default.createElement("i", null,
                            "(",
                            method.result.name,
                            ")"))),
                react_1.default.createElement(ContentDescriptor_1.default, { contentDescriptor: method.result, hideRequired: true, uiSchema: uiSchema })),
        method.errors && method.errors.length > 0 &&
            react_1.default.createElement("section", { key: "errors" },
                react_1.default.createElement("h2", null, "Errors"),
                react_1.default.createElement(Errors_1.default, { errors: method.errors, reactJsonOptions: reactJsonOptions })),
        method.examples && method.examples.length > 0 &&
            react_1.default.createElement("section", { key: "examples" },
                react_1.default.createElement("div", { style: { marginBottom: "var(--ifm-heading-margin-bottom)" } },
                    react_1.default.createElement("h2", { style: { marginRight: '3px' } }, method.examples.length > 1 ? "Examples" : "Example")),
                react_1.default.createElement(ExamplePairings_1.default, { uiSchema: uiSchema, components: components, examples: method.examples, onExamplePairingChange: onExamplePairingChange, method: method, reactJsonOptions: reactJsonOptions })),
        links && links.length > 0 &&
            react_1.default.createElement("section", { key: "links-title" },
                react_1.default.createElement("h2", null, "Links"),
                react_1.default.createElement("section", { key: "links" },
                    react_1.default.createElement(Links_1.default, { links: links, components: components }))),
        methodPlugins && methodPlugins.length > 0 &&
            react_1.default.createElement("section", { key: "method-plugins" }, methodPlugins.map((CompDef) => {
                return (react_1.default.createElement(CompDef, { openrpcMethodObject: method }));
            }))));
};
exports.default = Method;
